src/translate.py

```diff
@@ -3168,11 +3197,17 @@ def assign_phis(used_phis: List[PhiExpr], stack_info: StackInfo) -> None:
         phi = used_phis[i]
         assert phi.num_usages > 0
         assert len(phi.node.parents) >= 2
-        exprs = []
+
+        # Group parent nodes by the value of their phi register
+        equivalent_nodes: Dict[Expression, List[Node]] = {}
```

Comment:
`= defaultdict(list)` if you want to avoid the `if expr not in equivalent_nodes` check